Fix failure when path contains whitespace

EarlWare opened the pull request:
This commit changes how the `dbfile` string is created.  Rather than using `{URL}.String()` in `net/url` (in which, paths get escaped so whitespaces are replaced with `%20`), this change uses `{URL}.Path` field (which is not escaped, so whitespaces are preserved).      This is my attempt at a fix for issue  #499 (which seems prematurely closed, as its still an issue).

dhui commented on database/sqlite3/sqlite3.go:
```diff
@@ -96,12 +95,13 @@ func (m *Sqlite) Open(url string) (database.Driver, error) {
 		return nil, err
 	}
-	dbfile := strings.Replace(migrate.FilterCustomQuery(purl).String(), "sqlite3://", "", 1)
+	dbfile := purl.Path
```
We can't use `Path` directly since [options](https://github.com/mattn/go-sqlite3#dsn-examples) need to be passed to the underlying sqlite db driver

EarlWare replied:
That explains why that was using .String() in the first place...! 

So it looks like adding in `+ migrate.FilterCustomQuery(purl).Query().Encode()` should have the same net effect without escaping the path portion of the URL.   I've added the change and rebased to squash the commits into one.

PS. Thanks for the speedy review!  I didn't expect to see any movement on this today!

EarlWare replied:
Nope, I misread .Encode(), will need some logic to ensure it is either "?" + encodedQuery, or nothing.  Making another revision right now.

dhui reviewed (changes requested):
Thanks for the PR!

EarlWare commented:
Added in a function to covert the URL to a string which takes into account the query string.   Also added in test cases to ensure that the function produces the expected results based on a few different inputs, including a url with a query, a url without a query, and a url with a path that includes whitespaces.
